tb_pulumi/cloudwatch.py

```diff
@@ -258,15 +260,17 @@ def __init__(
         )
         memory_utilization_enabled = memory_utilization_opts['enabled']
         del memory_utilization_opts['enabled']
-        memory_utilization = resource.name.apply(
-            lambda res_name: aws.cloudwatch.MetricAlarm(
+        memory_utilization = pulumi.Output.all(res_name=resource.name, cluster_arn=resource.cluster).apply(
+            lambda outputs: aws.cloudwatch.MetricAlarm(
```

Comment:
Should this be singular since it's acting on single output out of the all.apply, or is this a list of outputs?

Reply:
`pulumi.Output.all` takes either a list or a dict of resources. When those resources' promised values become available, the lambda runs, and `outputs` is either a list or a dict of values, depending on the input. So either way, `outputs` should represent a plurality of values.

There is a different way of calling this if you only have one value you need to wait on, and that is just that output's `apply` function, like `resource.arn.apply`. So you wouldn't use this form of waiting on outputs to receive a singular value.